### ISSUE TITLE
Question about internal way to compute settlement plan

### ISSUE THREAD

**Issue by eneiluj** [spiral-project/ihatemoney#437]
Hey! I'm trying to get the settlement plan in MoneyBuster to be identical with the one in IHateMoney. I made a pretty identical copy of the algorithm and I get different transactions in result. I suspect the member list order is not the same which leads to different results.

At the top of `models.py` I see
```
members = db.relationship("Person", backref="project")
```
and I can't figure out how this is ordered. In MoneyBuster, I tried to order the member list by `LOWER(name)`, by `name` and by `id` without success.

Any idea?

**Comment by truenicoco** [comment]
For reference, I am the annoying user that [reported the difference](https://gitlab.com/eneiluj/moneybuster/issues/15). On the gitlab issue tracker you can see the differences in the settlement plans, along with eneiluj unsuccessful (but promising) attempt to reproduce IHM's behaviour. Maybe this will help…

**Comment by almet** [comment]
Would this help you? @eneiluj

**Comment by eneiluj** [comment]
Yes it would. I still need to know if this sorting is case sensitive or not.

Thanks for your answer!

**Comment by almet** [comment]
I think it's case sensitive by default, but can't find any documentation to backup that statement…

**Comment by eneiluj** [comment]
Ok well I could check by myself by comparing settlements results. Thanks.

**Comment by Natim** [comment]
@eneiluj While doing elm-ihatemoney I discovered that there were a bug in the Python implementation, the bug make the settlement page to crash in certains cases.

It would be great to fix the code of the Python code and maybe use the lib that @almet did a while ago to handle this.

**Comment by eneiluj** [comment]
Well i ported the code in Php and in Java so I might have copied the bug too :smile:.

Did you identify in which case the crash happens?

**Comment by almet** [comment]
It's related to https://github.com/spiral-project/ihatemoney/issues/431, and the current implementation still has the bug.

I believe the correct way to fix this would be to use the same algorithm that we used in elm-ihatemoney, that I ported to python in the debts library.

That would mean the implementation would have to be changed in php and Java too, which makes it a (small) piece of work. I believe it's worth it, and the code itself [is pretty straightforward](https://framagit.org/almet/debts/blob/master/debts/solver.py)
